src/components/Welcome.vue

```diff
@@ -29,19 +57,41 @@
 import {
 	AppContent,
 } from '@nextcloud/vue'
+import { createNote } from '../NotesService'
+import { getDefaultSampleNote } from '../Util'
+import { generateFilePath } from '@nextcloud/router'
 
 export default {
 	name: 'Welcome',
 
 	components: {
 		AppContent,
 	},
+
+	methods: {
+		onNewNote() {
+			createNote('')
+				.then(note => {
+					const query = { new: getDefaultSampleNote() }
+					this.$router.push({
+						name: 'note',
+						params: { noteId: note.id.toString() },
+						query,
+					})
+				})
+				.catch(() => {
+				})
```

Comment:
I don't like this approach, because it generates a very long URI. Instead, you should add a second parameter to `createNote` that takes the content and directly sends it to the server in the same request that creates the new note.

Reply:
Sidebar? What do you mean?

But yes, please do not put a whole note in a query parameter. :-)